subsys/bluetooth/host/iso.c

```diff
@@ -1096,3 +1105,442 @@ int bt_iso_chan_send(struct bt_iso_chan *chan, struct net_buf *buf)
+	 * broadcaster and receiver (even if the device is both!)
+	 */
+
+	for (int i = 0; i < ARRAY_SIZE(bigs); i++) {
+		if (!bigs[i]) {
+			return i;
+		}
+	}
+
+	BT_DBG("Could not allocate any more BIGs");
+
+	return BT_ISO_INVALID_BIG_HANDLE;
+}
+
+static void cleanup_big(struct bt_iso_big *big)
+{
+	for (int i = 0; i < big->num_bis; i++) {
+		bt_iso_cleanup(big->bis[i]->conn);
+	}
+	big->handle = BT_ISO_INVALID_BIG_HANDLE;
+}
+
+static void big_disconnect(struct bt_iso_big *big)
+{
+	for (int i = 0; i < big->num_bis; i++) {
+		bt_iso_disconnected(big->bis[i]->conn);
+	}
+}
+
+static int big_init_bis(struct bt_iso_big *big, bool broadcaster)
+{
+	big->handle = get_free_big_handle();
+
+	if (big->handle == BT_ISO_INVALID_BIG_HANDLE) {
+		return -ENOMEM;
+	}
+
+	for (int i = 0; i < big->num_bis; i++) {
+		struct bt_iso_chan *bis = big->bis[i];
+
+		if (!bis) {
+			BT_DBG("BIS was NULL");
+			return -EINVAL;
+		}
+
+		if (bis->conn) {
+			BT_DBG("BIS conn was already allocated");
+			return -EALREADY;
+		}
+
+		if (!bis->qos ||
+		    bis->qos->dir != (broadcaster ? BT_ISO_CHAN_QOS_OUT
+						  : BT_ISO_CHAN_QOS_IN)) {
+			BT_DBG("BIS QOS was invalid");
+			return -EINVAL;
+		}
+
+		bis->conn = iso_new();
+
+		if (!bis->conn) {
+			return -ENOMEM;
+		}
+
+		bis->conn->iso.big_handle = big->handle;
+		bis->conn->iso.is_bis = true;
+		bis->conn->iso.bis_id = bt_conn_index(bis->conn);
+
+		bt_iso_chan_add(bis->conn, bis);
+	}
+
+	return 0;
+}
+
+static int hci_le_create_big(struct bt_le_ext_adv *padv,
+			     struct bt_iso_big_create_param_t *param)
+{
+	struct bt_hci_cp_le_create_big *req;
+	struct bt_hci_evt_le_big_complete *evt;
+	struct bt_iso_big *big = param->big;
+	struct net_buf *buf;
+	struct net_buf *rsp;
+	int err;
+	static struct bt_iso_chan_qos *qos;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_CREATE_BIG, sizeof(*req));
+
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	qos = param->big->bis[0]->qos;
+
+	req = net_buf_add(buf, sizeof(*req));
+	req->big_handle = big->handle;
+	req->adv_handle = padv->handle;
+	req->num_bis = big->num_bis;
+	sys_put_le24(qos->interval, req->sdu_interval);
+	req->max_sdu = sys_cpu_to_le16(qos->sdu);
+	req->max_latency = sys_cpu_to_le16(qos->latency);
+	req->rtn = qos->rtn;
+	req->phy = qos->phy;
+	req->packing = qos->packing;
+	req->framing = qos->framing;
+	req->encryption = param->encryption;
+	if (req->encryption) {
+		memcpy(req->bcode, param->bcode, sizeof(req->bcode));
+	} else {
+		memset(req->bcode, 0, sizeof(req->bcode));
+	}
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_CREATE_BIG, buf, &rsp);
+	if (err) {
+		return err;
+	}
+
+	evt = (struct bt_hci_evt_le_big_complete *)rsp->data;
+	if (evt->status ||
+	    evt->big_handle != big->handle ||
+	    evt->num_bis != big->num_bis) {
+		err = -EIO;
+	} else {
+		for (int i = 0; i < big->num_bis; i++) {
+			big->bis[i]->conn->handle = evt->handle[i];
+			bt_iso_chan_set_state(big->bis[i], BT_ISO_CONNECTED);
+		}
+
+		bigs[big->handle] = big;
+	}
```

Comment:
Here `rsp` is command response, not LE Create BIG Complete event.
LE Create BIG has a command status followed by an asynchronous LE Create BIG Complete event.